runtime/criusupport/criusupport.c

```diff
@@ -352,19 +370,34 @@ Java_org_eclipse_openj9_criu_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
 		if (directoryBuf != directoryChars) {
 			j9mem_free_memory(directoryChars);
 		}
-		if (NULL != currentThread->currentException) {
-			currentExceptionLocalRef = (jthrowable) vmFuncs->j9jni_createLocalRef(env, currentThread->currentException);
-			currentThread->currentException = NULL;
-
-			if (NULL == currentExceptionLocalRef) {
-				/* nothing else to do */
-				vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
-			}
-		}
 
 		vmFuncs->internalExitVMToJNI(currentThread);
 #endif /* defined(LINUX) */
 	}
 
-	return constructResult(env, resultType, currentExceptionLocalRef);
+	if (NULL != currentExceptionClass) {
+		msgCharLength = j9str_printf(PORTLIB, NULL, 0, nlsMsgFormat, systemReturnCode);
+		exceptionMsg = (char *)j9mem_allocate_memory(msgCharLength, J9MEM_CATEGORY_VM);
+
+		j9str_printf(PORTLIB, exceptionMsg, msgCharLength, nlsMsgFormat, systemReturnCode);
+
+		jmethodID init = NULL;
+		if (vm->criuJVMCheckpointExceptionClass == currentExceptionClass) {
+			init = vm->criuJVMCheckpointExceptionInit;
+		} else if (vm->criuSystemCheckpointExceptionClass == currentExceptionClass) {
+			init = vm->criuSystemCheckpointExceptionInit;
+		} else {
+			init = vm->criuRestoreExceptionInit;
+		}
+		jstring jExceptionMsg = (*env)->NewStringUTF(env, exceptionMsg);
```

Comment:
This needs an exception check between this call the NewObject call or we may be in an inconsistent jni state (ie: calling back in with a pending exception)